docs/developer/pull-requests.md

```diff
@@ -0,0 +1,122 @@
+# Pull Request guidelines
+This document outlines some good practices to follow when creating or reviewing PRs. These are all guidelines and not hard rules, so take them as so.
+
+## As an author
+
+### PR size
+
+The recommended maximum lines of change is 500. Split PRs as necessary and be guided by the Single Responsibility Principle. Put yourself in the reviewers shoes and consider the cognitive load on the reviewer.
+
+For example, a PR with 1000 LoC touching 20 files could be easy to review if the changes are grammar or could be difficult to review if there are functional changes in loosely related components.
+
+### PR descriptions
```

Comment:
I've felt the pain of poorly constructed messages and over reliance on automated squash features when needing to become a git historian, but I've never found a way to get people to adopt it.

Trying to get people to adopt a good practice for an OSS project with random contributions can be even more of a challenge.

If we do want guidelines (which I think we have to because they're tricky to enforce) I suggest regurgitating this reference (my goto): https://cbea.ms/git-commit/

Reply:
In the past I've used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) with tooling like [clog](https://github.com/clog-tool/clog-cli) or [changelog](https://github.com/skuid/changelog) for validation and changelog generation. Not required for this review, but I certainly wouldn't be opposed to adopting something like that

Reply:
If we're going to commit to auto generated change logs then conventional commits or equivalent can be super useful - release preparation can be time consuming.

Conventional commits compliment the guidelines I posted.

Reply:
My 2c on the topic:
Standardized commit messages are very useful and I would be in favor of adopting a standard. Conventional commits works for me but I don't have a strong preference for any standard in particular. I do think that if we do this, it should go hand by hand with some kind of automation, at least for validation.

However, in my opinion, the content of the commit messages matters more than the format itself. And I believe that's where we can provide more value: on guidelines for folks on how to write good commit messages. Maybe this is where the article Chris linked can help. By no means I think this is incompatible with a standardized format, I just think we wouldn't do a lot by only agreeing on a format without trying to improve the content of the messages.

In addition, this topic resurfaces the discussion about manually squashing commits before merge. In my opinion, things like `* fix typos` or `* pass linter` don't add anything useful to the final commit message. 

That said, since I think both topics require further discussion and they have some ramifications in our processes (automation, changing the merge process to manually squash commits...), I would like to start a separate conversation and allow this PR to move forward. Once we reach agreement on the commit message topic, we can just update this doc (it's supposed to be a living document).

@micahhausler Added the clarification about this applying only to github pull requests